app/services/application/ApplicationService.scala

```diff
@@ -31,18 +36,41 @@ import scala.concurrent.{ ExecutionContext, Future }
 object ApplicationService extends ApplicationService {
   val appRepository = applicationRepository
   val eventService = EventService
+  val pdRepository = faststreamPersonalDetailsRepository
+  val cdRepository = faststreamContactDetailsRepository
 }
 
 trait ApplicationService extends EventSink {
   implicit val ec: ExecutionContext = scala.concurrent.ExecutionContext.Implicits.global
   val appRepository: GeneralApplicationRepository
+  val pdRepository: PersonalDetailsRepository
+  val cdRepository: ContactDetailsRepository
+
+  val Candidate_Role = "Candidate"
 
   def withdraw(applicationId: String, withdrawRequest: WithdrawApplication)
-    (implicit hc: HeaderCarrier, rh: RequestHeader): Future[Unit] = eventSink {
-    appRepository.withdraw(applicationId, withdrawRequest) map { _ =>
-      DataStoreEvents.ApplicationWithdrawn(applicationId, withdrawRequest.withdrawer) ::
-      AuditEvents.ApplicationWithdrawn(Map("applicationId" -> applicationId, "withdrawRequest" -> withdrawRequest.toString)) ::
-      Nil
-    }
+    (implicit hc: HeaderCarrier, rh: RequestHeader): Future[Unit] = {
+
+    appRepository.find(applicationId).flatMap{
+      case Some(candidate) =>
+        for {
+          cd <- cdRepository.find(candidate.userId)
+        } yield {
```

Comment:
Yielding a future in this yield block causes a Future[Future[Unit] to be returned. Once this gets to `appRepository.find(applicationId).flatMap{`, it will be mapped onto another future, but the flatmapping only flattens one level, so you still end up with Future[Future[Unit]]. Sad times. This means that failures will not necessarily be caught